src/components/disclosure/disclosure.module.css

```diff
@@ -0,0 +1,98 @@
+.root {
+  background-color: white;
+  border-radius: 6px;
+  width: 100%;
+}
+
+.root-expanded {
+  composes: hds-surface-mid from global;
+}
+
+.button {
+  composes: hds-surface-mid from global;
+  align-items: center;
+  background-color: transparent;
+  border-radius: 6px;
+  border: none;
+  cursor: pointer;
+  display: flex;
+  padding: 16px;
+  text-align: left;
+  width: 100%;
+
+  & svg {
+    color: var(--token-color-neutral-foreground-faint);
+
+    /* Only enable animation if query is supported and value is no-preference */
+    @media (prefers-reduced-motion: no-preference) {
+      transition: transform 0.2s;
+    }
+  }
+
+  /* aka Disclosure is open */
+  &[aria-expanded='true'] {
+    box-shadow: none;
+
+    & svg {
+      transform: rotate(90deg);
+    }
+  }
+
+  /* aka Disclosure is closed */
+  &[aria-expanded='false'] {
+    &:hover {
+      box-shadow: var(--token-surface-high-box-shadow);
+      cursor: pointer;
+    }
+  }
+
+  /*
+  TODO: not composing focus styles here because they use a border. Long term
+  goal is to refactor those to use a box-shadow only like in this situation.
+  */
+  &:focus,
+  &:focus-visible {
+    outline: none;
+  }
+
+  &:focus-visible {
+    box-shadow: var(--token-focus-ring-box-shadow);
+  }
+}
+
+/*
+Notes:
+  - Setting only the default state of the .content container prevents us from
+    accidentally setting the open state of the container to something incorrect
+    that can potentially break the semantics of the element
+  - Using the aria-expanded attribute selector ensures we're correctly setting
+    the value in the component's JS
+  - Using the next-sibling combinator (`+`) ensures that the .content container
+    is the sibling element immediately after the button
```

Comment:
🥇

Reply:
Thanks for the validation, I was hoping this information would be helpful. 😊